common/libraries/util/pow_utilities.c

```diff
@@ -88,18 +88,16 @@ void pow_get_approx_time_in_secs(const uint8_t target[SHA256_SIZE], uint32_t* ti
         }
     }
 
-    if ((256U - number_bits_set) > 31U) {
+    if ((256U - number_bits_set) > 63U) {
         // This is too long of a time to calculate, just return
-        *time_in_secs_out = 99U;
+        *time_in_secs_out = UINT32_MAX;
         return;
     }
 
-    uint32_t time_in_secs = 1U;
-    time_in_secs = (uint32_t)(time_in_secs << (256U - number_bits_set)) / pow_hash_rate;
+    uint64_t time_in_secs = 1U;
+    time_in_secs = (time_in_secs << (256U - number_bits_set)) / pow_hash_rate;
```

Comment:
This being an important calculation, we can also add an assert for `pow_hash_rate`

Reply:
```suggestion
    ASSERT(pow_hash_rate != 0);
    time_in_secs = (time_in_secs << (256U - number_bits_set)) / pow_hash_rate;
```